### DISCUSSION TITLE
S3 predicate pushdown

### DISCUSSION THREAD

**Discussion by jeyarajcs** [duckdb/duckdb#15026] [Q&A]
I am trying to understand, How Duckdb works with S3(parquet files) for Automatic Filter & Projection Pushdown(https://duckdb.org/2021/06/25/querying-parquet.html#automatic-filter--projection-pushdown).

I have around 100 parquet files(Each one 500Mb, 50+ columns) in S3
If I select specific columns, I know duckdb skips the other columns and only the selected columns will be loaded into Memory.  What about filters? Can DuckDB filters the rows from S3 files that influences the query without loading all the rows into memory?

**Comment by Tishj** [comment]
Yes, it uses the parquet metadata for this
